src/meta/src/barrier/mod.rs

```diff
@@ -272,19 +305,67 @@ impl CheckpointControl {
     ) {
         let timer = self.context.metrics.barrier_latency.start_timer();
 
+        let mut newly_finished_table_ids = HashSet::new();
+
+        for (table_id, creating_streaming_job) in &mut self.creating_streaming_job_controls {
+            match creating_streaming_job.status {
+                CreatingStreamingJobStatus::ConsumingSnapshot { .. }
+                | CreatingStreamingJobStatus::ConsumingLogStore { .. } => {}
+                CreatingStreamingJobStatus::Finishing(prev_epoch) => {
+                    if command_ctx.kind.is_checkpoint()
+                        && prev_epoch != command_ctx.prev_epoch.value().0
```

Comment:
Why can we transition the job status from Finishing to Finished on `prev_epoch != command_ctx.prev_epoch.value().0`? I thought the transitioning should be done in finish collecting the `prev_epoch` in `Finishing`.

Reply:
The status actually reflect the status after injecting the latest barrier and does not directly relate to collecting barriers. For example, even though we have entered the `Finishing` status after injecting a barrier, theoretically there can be still some uncollected barriers injected at status `ConsumingSnapshot`.

BTW, the naming of the status was confusing. I have changed the original `Finishing` and `Finished` to `ConsumingUpstream` and `Finishing`.